### ISSUE TITLE
NE3: Segfaults / aborts 

### ISSUE THREAD

**Issue by itdaniher** [pydata/numexpr#254]
```
import numexpr3 as ne
import numpy as np

for n in [2**i for i in range(0,20)]:
    print(n)
    a = (np.random.random(n)+np.random.random(n)*1j).astype('complex64')
    b = ne.evaluate("b = 1+a")
```

```*** Error in `python': corrupted double-linked list: 0x0078ece0 *** ```

on Python3.5.2 / ARMHF.

```
Unallocated output: b broadcast shape: (1,), dtype: complex64
*** Error in `python': free(): invalid pointer: 0x00007f7a733b1bc8 ***
*** Error in `python': munmap_chunk(): invalid pointer: 0x*** Error in `python': munmap_chunk(): invalid pointer: 0x00007f7a733be110 ***
[1]    4146 segmentation fault  python ne3.py
```

On Python3.6.0 / X86_64

**Comment by itdaniher** [comment]
On Python3.6.0 / X86_64 the tests segfault before the first one completes.

On Python3.5.2 / ARMHF the tests run until ```TypeError: don't know how to make test from: 3.0.1a0  ``` - other results at http://sprunge.us/CEBj.

**Comment by robbmcleod** [comment]
There's something up with Python 3.6 at present.  I'm not sure what it is, but I found problems with it calling Windows standard functions, so whatever the problem is there it's likely beyond my ability to fix.

For Python 3.5, I see that you're using magic output but changing the array sizes with each iteration.  I'll have to think about how to to generate an error in this case, or handle it efficiently without impacting performance too much.  Probably it needs some checks in the signature for array sizes. Practically if you're calling NE3 within a loop you should use the object oriented interface.

Generally the nice way to see what's happening is:

    import numexpr3 as ne
    import numpy as np
   
    for n in [2**i for i in range(0,20)]:
        print(n)
        a = (np.random.random(n)+np.random.random(n)*1j).astype('complex64')
        neObj = ne.NumExpr( 'b = 1+a' )
        neObj.disassemble()
        neObj.run( check_arrays=True )

The `disassemble` method is fairly verbose now.

**Comment by itdaniher** [comment]
@robbmcleod Very cool, thanks!

**Comment by robbmcleod** [comment]
Should be fixed by 07d9245.
